realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -478,11 +478,11 @@ private void emitCopyOrUpdateMethod(JavaWriter writer) throws IOException {
                 className, // Return type
                 "copyOrUpdate", // Method name
                 EnumSet.of(Modifier.PUBLIC, Modifier.STATIC), // Modifiers
-                "Realm", "realm", className, "object", "boolean", "update" // Argument type & argument name
+                "Realm", "realm", className, "object", "boolean", "update", "Map<RealmObject,RealmObject>", "cache" // Argument type & argument name
```

Comment:
Shouldn't it be `Map<? extends RealmObject, ? extends RealmObject>` ?
Also, since now we are at a point where it would be nice to distinguish proxy objects from standalone objects, wouldn't it maybe make sense to have proxy objects implement an empty interface, for generics purposes?